volmdlr/utils/intersections.py

```diff
@@ -97,3 +99,97 @@ def ellipse2d_line_intersections(ellipse2d, line2d):
     if point1 == point2:
         return [point1]
     return [point1, point2]
+
+
+def get_circle_intersections(circle1, circle2):
+    """
+    Calculates the intersections between two circle 2d.
+
+    :param circle1: circle 1 verify intersection with bspline
+    :param circle2: circle 2 to search for intersections.
+    :return: a list with all intersections between the two circles.
+    """
+    x0, y0 = circle1.center
+    x1, y1 = circle2.center
+
+    d = math.sqrt((x1 - x0) ** 2 + (y1 - y0) ** 2)
+
+    # non-intersecting
+    if d > circle1.radius + circle2.radius:
+        return []
+    # One circle within other
+    if d < abs(circle1.radius - circle2.radius):
+        return []
+    # coincident circles
+    if d == 0 and circle1.radius == circle2.radius:
+        return []
+    a = (circle1.radius ** 2 - circle2.radius ** 2 + d ** 2) / (2 * d)
+    h = math.sqrt(circle1.radius ** 2 - a ** 2)
+    x2 = x0 + a * (x1 - x0) / d
+    y2 = y0 + a * (y1 - y0) / d
+    x3 = x2 + h * (y1 - y0) / d
+    y3 = y2 - h * (x1 - x0) / d
+
+    x4 = x2 - h * (y1 - y0) / d
+    y4 = y2 + h * (x1 - x0) / d
+
+    return [volmdlr.Point2D(x3, y3), volmdlr.Point2D(x4, y4)]
+
+
+def bspline_intersections_initial_conditions(edge2d, bsplinecurve2d):
```

Comment:
```suggestion
def bspline_intersections_initial_conditions(edge2d, bsplinecurve2d, resolution: int = 10):
```

Reply:
By adding resolution into the parameters, I think then it will be easier to find an empirical optimal value to algorithm performance and accuracy, if needed in futures development. What do you think?